### PR TITLE
Siamese network classifier on Alzheimer’s disease of the ADNI brain data - Fanhao Zeng - 46398086

### PR DESCRIPTION
**Pull request by FanHaoZeng**
### **Description:**
This is a solution of classify Alzheimer's disease on the ADNI dataset with Siamese network and have an 82% Accuracy on the test set with 50 epochs training. 

### **Files overview:**

 Images:
- Contain the required images for README.md file.

recognition
- dataset.py: contain the code to load the data and some required functions.
- modules.py: contain the main structure of the model, and functions like forward
- Train: contain the training and validation operations, whole training process take place in main()
- predict: use to test the performance of the trained model
- VGG: an alternate subnetwork for Siamese, can replace resnet18

### **Performance**
The model has been trained 50 epochs with Adam optimizer and have **82%** accuracy on the test set.

Future work may include optimize network structure, reduce the training cost and find a more efficient way to load data.

### **Author**
Fanhao Zeng
46390806

Best wishes.

### CODE REVIEW

**Comment by FanHaoZeng** [comment]
Sorry, I have accidently done "git add *” and "git commit" which led me upload the logs and the model. I will update the file Structure here.
results:  contain all the trained model.
recognition-> logs/siamese_experiment: contain the training logs.

**Comment by nathasha-naranpanawa** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme: OK
- some information provided on the algorithm
- training and testing details are vague
- loss and accuracy plots provided for the siamese, but only for the training set, not very useful

Commit messages: OK

Code: 
- design is ok
- well commented
- only implemented a siamese network, no classifier

Functionality/Performance: 
- **only performing similarity detection between pairs of images, no classification of single images**
- achieves good accuracy for similarity detection with the siamese

General comments:
- **algorithm solves the respective problem only half-way**
- you need to have added a classifier that can classify an individual image as AD or NC with their features extracted by the trained siamese network

**Comment by FanHaoZeng** [comment]
The classification of single images functions has been added.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem half way -3
Driver Script present
File structure present but incorrect, create folder for your task with name -2
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, fix directory structure to allow merge -2
Request Description good

**Comment by shakes76** [comment]
Fix the directory structure to recieve feedback +2 marks

**Comment by FanHaoZeng** [comment]
Hi Shakes,
Thank you for your review, I have fixed the directory structure. :)

Best wishes, 
Fanhao Zeng

**Comment by wangzhaomxy** [comment]
Feedback attempt and feedback marks granted +2.
